supervisor/dbus/network/configuration.py

```diff
@@ -63,6 +63,7 @@ class IpProperties:
     """IP properties object for Network Manager."""
 
     method: str | None
+    addr_gen_mode: int | None
```

Comment:
This isn't really accurate typing. It is always an `int` for ipv6 and always `None` for ipv4. I think we should make a new `Ip6Properties` which inherits from `IpProperties` and adds this field. And then use `IpProperties` in ipv4 places and `Ip6Properties` in ipv6 places.